src/features/defi/providers/foxy/components/FoxyManager/Deposit/components/Approve.tsx

```diff
@@ -50,39 +57,57 @@ export const Approve: React.FC<StepComponentProps> = ({ onNext }) => {
   // user info
   const { state: walletState } = useWallet()
 
-  if (!state || !dispatch) return null
+  const getDepositGasEstimate = useCallback(
+    async (deposit: DepositValues) => {
+      if (!state?.userAddress || !assetReference || !api) return
+      try {
+        const [gasLimit, gasPrice] = await Promise.all([
+          api.estimateDepositGas({
+            tokenContractAddress: assetReference,
+            contractAddress,
+            amountDesired: bnOrZero(deposit.cryptoAmount)
+              .times(`1e+${asset.precision}`)
+              .decimalPlaces(0),
+            userAddress: state.userAddress,
+          }),
+          api.getGasPrice(),
+        ])
+        return bnOrZero(gasPrice).times(gasLimit).toFixed(0)
+      } catch (error) {
+        moduleLogger.error(
+          { fn: 'getDepositGasEstimate', error },
+          'Error getting deposit gas estimate',
+        )
+        toast({
+          position: 'top-right',
+          description: translate('common.somethingWentWrongBody'),
+          title: translate('common.somethingWentWrong'),
+          status: 'error',
+        })
+      }
+    },
+    [api, asset.precision, assetReference, contractAddress, state?.userAddress, toast, translate],
+  )
 
-  const getDepositGasEstimate = async (deposit: DepositValues) => {
-    if (!state.userAddress || !assetReference || !api) return
-    try {
-      const [gasLimit, gasPrice] = await Promise.all([
-        api.estimateDepositGas({
-          tokenContractAddress: assetReference,
-          contractAddress,
-          amountDesired: bnOrZero(deposit.cryptoAmount)
-            .times(`1e+${asset.precision}`)
-            .decimalPlaces(0),
-          userAddress: state.userAddress,
-        }),
-        api.getGasPrice(),
-      ])
-      return bnOrZero(gasPrice).times(gasLimit).toFixed(0)
-    } catch (error) {
-      moduleLogger.error(
-        { fn: 'getDepositGasEstimate', error },
-        'Error getting deposit gas estimate',
-      )
-      toast({
-        position: 'top-right',
-        description: translate('common.somethingWentWrongBody'),
-        title: translate('common.somethingWentWrong'),
-        status: 'error',
-      })
-    }
-  }
+  const accountFilter = useMemo(() => ({ accountId: accountId ?? '' }), [accountId])
+  const accountMetaData = useAppSelector(state =>
+    selectPortfolioAccountMetadataByAccountId(state, accountFilter),
+  )
+
+  const bip44Params = accountMetaData?.bip44Params
```

Comment:
https://github.com/shapeshift/web/blob/feat_foxy_multi_account/src/state/slices/portfolioSlice/selectors.ts#L154

if you want to

Reply:
Nice one ser, way cleaner! 2858d2f4ba9e1ec7ee2c948ac5b8a961e24503b1